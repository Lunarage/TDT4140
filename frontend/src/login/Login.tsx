import React, { useState } from "react";
import { useDispatch } from 'react-redux';
import styled from "styled-components";
import Button from "../components/Button";
import Header from "../components/Header";
import InputField from "../components/InputField";
import { baseUrl } from '../consts';
import HttpClient from '../utilities/HttpClient';
import WelcomeLogo from "../welcome/WelcomeLogo";
import { getUser } from '../store/actionCreators';

const PageWrapper = styled.div`
  display: flex;
  flex-direction: column;
  background-image: url(/static/EverestTiny.png);
  background-repeat: no-repeat;
  height: 100vh;
  background-size: auto 100%;
`;

const WelcomeLogoWrapper = styled.div`
  margin-left: 280px;
  margin-top: 35px;
`;

const LoginWrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 500px;
  height: 500px;
  position: absolute;
  right: 10%;
  top: 170px;
`;

const ButtonsWrapper = styled.div`
  width: 100%;
  display: flex;
  justify-content: space-evenly;
`;

const LoginWidget = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  min-height: 300px;
  width: 100%;
  margin: 20px;
  border: hidden;
  border-radius: 10px;
  box-shadow: rgba(0, 0, 0, 0.45) 1px 5px 15px;
`;

const InputWrapper = styled.div`
  width: 300px;
  margin-top: 30px;
`;

const LoginHeader = styled.div`
  font-size: 20px;
  margin: 10px;
  font-weight: bold;
`;

export enum Method {
  login = "Logg inn",
  register = "Registrer",
}

const Login = () => {
  const dispatch = useDispatch();

  const [method, setMethod] = useState<Method>(Method.login);
  const [email, setEmail] = useState<string>();
  const [username, setUsername] = useState<string>();
  const [firstName, setFirstName] = useState<string>();
  const [lastName, setLastName] = useState<string>();
  const [password, setPassword] = useState<string>();
  const [confPassword, setConfPassword] = useState<string>();

  const handleToken = (token: string) => {

  }

  const handleSubmit = () => {
    let client = new HttpClient(baseUrl)
    if (method === Method.login && username && password) {
      dispatch(getUser(username, password))
      console.log(':)')
      console.log(username)
    }
    else {
      console.log(":(")
    }
  }

  return (
    <PageWrapper>
      <Header />
      <WelcomeLogoWrapper>
        <WelcomeLogo />
      </WelcomeLogoWrapper>
      <LoginWrapper>
        <LoginWidget>
          <LoginHeader>{method}</LoginHeader>
          <InputWrapper>
            {method === Method.register && (<> <InputField name="First name" onChangeFunc={(val) => setFirstName(val)} />
              <InputField name="Last name" onChangeFunc={(val) => setLastName(val)} />
              <InputField name="E-post" onChangeFunc={(val) => setEmail(val)} /> </>)}
            <InputField name="Brukernavn" onChangeFunc={(val) => setUsername(val)} />
            <InputField name="Passord" onChangeFunc={(val) => setPassword(val)} />
            {method === Method.register && (
              <InputField name="Confirm passord" onChangeFunc={(val) => setConfPassword(val)} />
            )}
          </InputWrapper>
          <Button text="Submit" onClickFunc={() => handleSubmit()} />
        </LoginWidget>
        <ButtonsWrapper>
          <Button
            text={Method.register}
            onClickFunc={() => setMethod(Method.register)}
          />
          <Button
            text={Method.login}
            onClickFunc={() => setMethod(Method.login)}
          />
        </ButtonsWrapper>
      </LoginWrapper>
    </PageWrapper>
  );
};

export default Login;
