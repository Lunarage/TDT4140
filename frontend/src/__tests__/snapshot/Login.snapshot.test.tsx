import { mount } from "enzyme";
import Enzyme from "enzyme";
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import toJSON from "enzyme-to-json";
import Login from '../../login/Login';
import { store } from '../..';
import { Provider } from 'react-redux';

Enzyme.configure({ adapter: new Adapter() });

jest.mock("react-router-dom", () => ({
  ...jest.requireActual('react-router-dom') as any,
  useHistory: () => ({
    push: jest.fn(),
  }),
}));

const LoginWrapper = () => (
  <Provider store={store}>
    <Login />
  </Provider>
);

describe("Login", () => {
  it("renders Login", () => {
    mount(<LoginWrapper />);
  });
  it("Should match Login snapshot", () => {
    const comp = mount(<LoginWrapper />);
    expect(toJSON(comp)).toMatchSnapshot();
  });
});

