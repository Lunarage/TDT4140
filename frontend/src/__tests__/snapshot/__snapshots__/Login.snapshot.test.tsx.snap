// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Should match Login snapshot 1`] = `
<LoginWrapper>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <Login>
      <styled.div>
        <div
          className="sc-iBaPrD AJnpx"
        >
          <Header>
            <styled.div>
              <div
                className="sc-iktFzd uGybn"
              >
                <styled.div
                  onClick={[Function]}
                >
                  <div
                    className="sc-cBNfnY bDxDMJ"
                    onClick={[Function]}
                  >
                    Gjør No'
                  </div>
                </styled.div>
                <styled.div>
                  <div
                    className="sc-gWHgXt bcglBl"
                  >
                    <styled.div
                      onClick={[Function]}
                    >
                      <div
                        className="sc-hiSbYr hZBClG"
                        onClick={[Function]}
                      >
                        Aktiviteter
                      </div>
                    </styled.div>
                    <styled.div
                      onClick={[Function]}
                    >
                      <div
                        className="sc-hiSbYr hZBClG"
                        onClick={[Function]}
                      >
                        Arrangementer
                      </div>
                    </styled.div>
                  </div>
                </styled.div>
                <styled.div
                  onClick={[Function]}
                >
                  <div
                    className="sc-jJEJSO cgHETT"
                    onClick={[Function]}
                  >
                    Logg inn
                  </div>
                </styled.div>
              </div>
            </styled.div>
          </Header>
          <styled.div>
            <div
              className="sc-eggNIi eWTXyJ"
            >
              <WelcomeLogo>
                <Styled(styled.div)>
                  <div
                    className="sc-cBNfnY sc-iUuytg bDxDMJ gjnBsR"
                  >
                    Gå ut
                  </div>
                </Styled(styled.div)>
                <Styled(styled.div)>
                  <div
                    className="sc-cBNfnY sc-bZSQDF bDxDMJ fKHsfl"
                  >
                    Gjør No'
                  </div>
                </Styled(styled.div)>
              </WelcomeLogo>
            </div>
          </styled.div>
          <styled.div>
            <div
              className="sc-cTkwdZ eBfFQS"
            >
              <styled.div>
                <div
                  className="sc-dOSReg ipFANJ"
                >
                  <styled.div>
                    <div
                      className="sc-cOajty kmBQZS"
                    >
                      Logg inn
                    </div>
                  </styled.div>
                  <styled.div>
                    <div
                      className="sc-bBrOnJ dDzzIR"
                    >
                      <InputField
                        name="Brukernavn"
                        onChangeFunc={[Function]}
                      >
                        <styled.div>
                          <div
                            className="sc-citwmv obwtd"
                          >
                            <styled.div>
                              <div
                                className="sc-jcVebW llCqVi"
                              >
                                Brukernavn
                              </div>
                            </styled.div>
                            <Input
                              onChange={[Function]}
                              type="text"
                            >
                              <div
                                className="ui input"
                              >
                                <input
                                  onChange={[Function]}
                                  type="text"
                                />
                              </div>
                            </Input>
                          </div>
                        </styled.div>
                      </InputField>
                      <InputField
                        name="Passord"
                        onChangeFunc={[Function]}
                        type="password"
                      >
                        <styled.div>
                          <div
                            className="sc-citwmv obwtd"
                          >
                            <styled.div>
                              <div
                                className="sc-jcVebW llCqVi"
                              >
                                Passord
                              </div>
                            </styled.div>
                            <Input
                              onChange={[Function]}
                              type="password"
                            >
                              <div
                                className="ui input"
                              >
                                <input
                                  onChange={[Function]}
                                  type="password"
                                />
                              </div>
                            </Input>
                          </div>
                        </styled.div>
                      </InputField>
                    </div>
                  </styled.div>
                   
                  <Button
                    onClickFunc={[Function]}
                    text="Submit"
                  >
                    <styled.div
                      onClick={[Function]}
                      style={
                        Object {
                          "backgroundColor": "#ff4e37",
                          "color": "white",
                          "minWidth": "8em",
                        }
                      }
                    >
                      <div
                        className="sc-bdfBwQ jzdiim"
                        onClick={[Function]}
                        style={
                          Object {
                            "backgroundColor": "#ff4e37",
                            "color": "white",
                            "minWidth": "8em",
                          }
                        }
                      >
                        <styled.div>
                          <div
                            className="sc-gsTCUz dfTHCv"
                          >
                            Submit
                          </div>
                        </styled.div>
                      </div>
                    </styled.div>
                  </Button>
                </div>
              </styled.div>
              <styled.div>
                <div
                  className="sc-jNMdTA dxOtOM"
                >
                  <Button
                    onClickFunc={[Function]}
                    text="Registrer"
                  >
                    <styled.div
                      onClick={[Function]}
                      style={
                        Object {
                          "backgroundColor": "#ff4e37",
                          "color": "white",
                          "minWidth": "8em",
                        }
                      }
                    >
                      <div
                        className="sc-bdfBwQ jzdiim"
                        onClick={[Function]}
                        style={
                          Object {
                            "backgroundColor": "#ff4e37",
                            "color": "white",
                            "minWidth": "8em",
                          }
                        }
                      >
                        <styled.div>
                          <div
                            className="sc-gsTCUz dfTHCv"
                          >
                            Registrer
                          </div>
                        </styled.div>
                      </div>
                    </styled.div>
                  </Button>
                  <Button
                    onClickFunc={[Function]}
                    text="Logg inn"
                  >
                    <styled.div
                      onClick={[Function]}
                      style={
                        Object {
                          "backgroundColor": "#ff4e37",
                          "color": "white",
                          "minWidth": "8em",
                        }
                      }
                    >
                      <div
                        className="sc-bdfBwQ jzdiim"
                        onClick={[Function]}
                        style={
                          Object {
                            "backgroundColor": "#ff4e37",
                            "color": "white",
                            "minWidth": "8em",
                          }
                        }
                      >
                        <styled.div>
                          <div
                            className="sc-gsTCUz dfTHCv"
                          >
                            Logg inn
                          </div>
                        </styled.div>
                      </div>
                    </styled.div>
                  </Button>
                </div>
              </styled.div>
            </div>
          </styled.div>
        </div>
      </styled.div>
    </Login>
  </Provider>
</LoginWrapper>
`;
